src/AlgoliaSearchCore.js

```diff
@@ -438,7 +421,62 @@ AlgoliaSearchCore.prototype._jsonRequest = function(initialOpts) {
     }
   }
 
-  var promise = doRequest(
+  function isCacheValidWithCurrentID() {
+    return client._useCache && cache && cache[cacheID] !== undefined;
```

Comment:
why did you make `cacheID` global instead of using it as a function parameter?

Reply:
I tried to stick at the pattern used in the code actually, I really don't like relying on global variable. I can change it but I might have some issues to name the variable since ESLint won't allow me to use the same the variable name (I think).

Reply:
```js
function isCacheValidWithCurrentID(currentCacheID) {
}
```

this would work right?

Reply:
Yes it would without the destructuring part.